sentry_dart_plugin/lib/src/configuration.dart

```diff
@@ -0,0 +1,155 @@
+    final pubspec = _getPubspec();
+    final config = pubspec['sentry_plugin'];
+
+    version = config?['release']?.toString() ?? pubspec['version'].toString();
+    name = pubspec['name'].toString();
+
+    uploadNativeSymbols = config?['upload_native_symbols'] ?? true;
+    uploadSourceMaps = config?['upload_source_maps'] ?? false;
+    includeNativeSources = config?['include_native_sources'] ?? false;
+
+    // uploading JS and Map files need to have the correct folder structure
+    // otherwise symbolication fails, the default path for the web build folder is build/web
+    // but can be customized so making it flexible.
+    final webBuildPath = config?['web_build_path']?.toString() ?? 'build/web';
+    webBuildFilesFolder = '$buildFilesFolder$_fileSeparator$webBuildPath';
+
+    // pro tip: add rule: stack.abs_path:org-dartlang-sdk* -app
+    // and: stack.abs_path:org-dartlang-sdk* -group
+    // to the issue grouping
+
+    project = config?['project']?.toString(); // or env. var. SENTRY_PROJECT
+    org = config?['org']?.toString(); // or env. var. SENTRY_ORG
+    wait = config?['wait'] ?? false;
+    authToken =
+        config?['auth_token']?.toString(); // or env. var. SENTRY_AUTH_TOKEN
+    logLevel =
+        config?['log_level']?.toString(); // or env. var. SENTRY_LOG_LEVEL
+
+    Log.taskCompleted(taskName);
+  }
+
+  void validateConfigValues() {
+    const taskName = 'validating config values';
+    Log.startingTask(taskName);
+
+    final environments = Platform.environment;
+
+    if (project.isNull && environments['SENTRY_PROJECT'].isNull) {
+      Log.errorAndExit('Project is empty, check \'project\' at pubspec.yaml');
+    }
+    if (org.isNull && environments['SENTRY_ORG'].isNull) {
+      Log.errorAndExit('Organization is empty, check \'org\' at pubspec.yaml');
+    }
+    if (authToken.isNull && environments['SENTRY_AUTH_TOKEN'].isNull) {
+      Log.errorAndExit(
+          'Auth Token is empty, check \'auth_token\' at pubspec.yaml');
+    }
+
+    try {
+      Process.runSync(cliPath!, ['help']);
+    } catch (exception) {
+      Log.errorAndExit(
+          'sentry-cli is not available, please follow https://docs.sentry.io/product/cli/installation/ \n$exception');
+    }
+
+    Log.taskCompleted(taskName);
+  }
+
+  /// Get the assets folder path from the .packages file
+  Future<void> _getAssetsFolderPath() async {
+    final packagesConfig = await loadPackageConfig(File(
+        '${Directory.current.path}$_fileSeparator.dart_tool${_fileSeparator}package_config.json'));
+
+    final packages = packagesConfig.packages
+        .where((package) => package.name == "sentry_dart_plugin");
+
+    if (packages.isNotEmpty) {
+      final path =
+          packages.first.packageUriRoot.toString().replaceAll('file://', '') +
+              'assets';
+
+      _assetsPath = Uri.decodeFull(path);
+    }
+
+    if (_assetsPath.isNull) {
+      Log.info('The assets folder is not found.');
+    }
+  }
+
+  void _findCliPath() {
```

Comment:
`l`: This finds and sets the cli path. I propose either renaming the function or splitting it up.